File: app/src/main/java/com/example/mrakopediareader/api/parser/WebsiteURLParser.java
package com.example.mrakopediareader.api.parser;

import android.net.Uri;

import com.example.mrakopediareader.api.dto.WebsiteUrl;

import org.json.JSONException;
import org.json.JSONObject;

import java.net.URI;

public class WebsiteURLParser extends ParserImpl<WebsiteUrl> {
    private static String KEY_TITLE = "title";
    private static String KEY_URL = "url";

    @Override
    public WebsiteUrl fromJsonObject(JSONObject object) throws JSONException {
        final String title = object.getString(KEY_TITLE);
        final Uri uri = Uri.parse(object.getString(KEY_URL));
        return new WebsiteUrl(title, uri);
    }
}

File: app/src/main/java/com/example/mrakopediareader/api/parser/WebsiteURLParser.kt
package com.example.mrakopediareader.api.parser

import android.net.Uri
import com.example.mrakopediareader.api.dto.WebsiteUrl
import org.json.JSONObject

internal class WebsiteURLParser : ParserImpl<WebsiteUrl>() {
    override fun fromJsonObject(jsonObject: JSONObject): WebsiteUrl {
        val title = jsonObject.getString(KEY_TITLE)
        val uri = Uri.parse(jsonObject.getString(KEY_URL))
        return WebsiteUrl(title, uri)
    }

    companion object {
        private const val KEY_TITLE = "title"
        private const val KEY_URL = "url"
    }
}